### ISSUE TITLE
Python 3.8: TypeError: 'type' object is not subscriptable

### ISSUE THREAD

**Issue by wico-silva** [frnmst/fpyutils#6]
The latest version (4.0.0) is giving me an error on Python 3.8.

```
   File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/md_toc/__init__.py", line 24, in <module>
    from .api import (
  File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/md_toc/api.py", line 31, in <module>
    import fpyutils
  File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/fpyutils/__init__.py", line 24, in <module>
    from . import exceptions, filelines, notify, path, shell
                  └ <module 'fpyutils.exceptions' from '/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/fpyutils/exceptions.py'>
  File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/fpyutils/filelines.py", line 36, in <module>
    keep_all_lines: bool = False) -> tuple[dict[int, int], str]:

TypeError: 'type' object is not subscriptable
```

I believe the problem is that `dict` is not a generic type on Python 3.8 unless you add `from __future__ import annotations`. Or, use `typing.Dict` instead.

Is this version still supporting Python 3.8 or only 3.9+?

**Comment by frnmst** [comment]
Thanks for reporting.

I bumped the minimum Python requirement from 3.5 to 3.8 to avoid a too big of a jump. Locally I'm running 3.11 so I haven't noticed this problem. I thought 3.8 was sufficient.

You can open a PR with the `from __future__ import annotations` solution.

In the meantime I need to setup tests for multiple Python versions, such as by using tox or something...

**Comment by frnmst** [comment]
I was able to use [asdf](https://github.com/asdf-vm/asdf) to install different Python versions.

```shell
. .venv/bin/activate \
	&& tox run-parallel \
	&& deactivate

py38: OK ✔ in 2.03 seconds
py39: OK ✔ in 2.06 seconds
py310: OK ✔ in 2.08 seconds
py37: OK ✔ in 2.1 seconds
py311: OK ✔ in 0.89 seconds
py312: OK ✔ in 1.11 seconds
  py37: OK (2.10=setup[1.69]+cmd[0.40] seconds)
  py38: OK (2.03=setup[1.64]+cmd[0.39] seconds)
  py39: OK (2.06=setup[1.66]+cmd[0.40] seconds)
  py310: OK (2.08=setup[1.67]+cmd[0.42] seconds)
  py311: OK (0.89=setup[0.50]+cmd[0.39] seconds)
  py312: OK (1.11=setup[0.69]+cmd[0.42] seconds)
  py313: OK (1.13=setup[0.74]+cmd[0.39] seconds)
  congratulations :) (3.27 seconds)
```

As you see it now works up to Python 3.7 without errors. There will be an update soon!

**Comment by frnmst** [comment]
This has been fixed in version 4.0.1.

Closing.
